olp-cpp-sdk-dataservice-read/src/ApiClientLookup.cpp

```diff
@@ -154,28 +154,36 @@ ApiClientLookup::ApiClientResponse ApiClientLookup::LookupApi(
 
   client::Condition condition{};
 
-  auto client = std::make_shared<client::OlpClient>();
-  client->SetSettings(std::move(settings));
-
   // when the network operation took too much time we cancel it and exit
   // execution, to make sure that network callback will not access dangling
   // references we protect them with atomic bool flag.
   auto flag = std::make_shared<std::atomic_bool>(true);
 
-  ApiClientLookup::ApiClientResponse api_response;
-  auto api_client_callback =
-      [&, flag](ApiClientLookup::ApiClientResponse response) {
-        if (flag->exchange(false)) {
-          api_response = std::move(response);
-          condition.Notify();
-        }
-      };
+  PlatformApi::ApisResponse api_response;
+  auto api_callback = [&, flag](PlatformApi::ApisResponse response) {
+    if (flag->exchange(false)) {
+      api_response = std::move(response);
+      condition.Notify();
+    }
+  };
 
   cancellation_context.ExecuteOrCancelled(
       [&, flag]() {
-        auto token = ApiClientLookup::LookupApiClient(
-            client, service, service_version, catalog,
-            std::move(api_client_callback));
+        const auto& base_url = GetDatastoreServerUrl(catalog.partition);
+
+        client::OlpClient client;
+        client.SetBaseUrl(base_url);
+        client.SetSettings(std::move(settings));
```

Comment:
There's usages of settings after move

Reply:
might be a problem in the future, now there is no movement, a function takes a const ref.